core/src/com/unciv/logic/automation/BattleHelper.kt

```diff
@@ -102,6 +102,13 @@ object BattleHelper {
         if (tileCombatant.getCivInfo() == combatant.getCivInfo()) return false
         if (!combatant.getCivInfo().isAtWarWith(tileCombatant.getCivInfo())) return false
 
+        if (combatant is MapUnitCombatant && combatant.unit.baseUnit.unitType.contains("Water") &&
+                combatant.unit.baseUnit.isMelee() && !combatant.unit.civInfo.hasUnique(UniqueType.LandUnitEmbarkation)){
+            if (tileCombatant is MapUnitCombatant && tileCombatant.unit.baseUnit.unitType.contains("Water")) {
```

Comment:
Btw I nested the ifs cause I thought it might look a bit more readable. But yes it should have been inverted, but I'll replace those with isWaterUnit and isLandUnit after I come back from faculty.

Reply:
Maybe yes to only let them attack if they can embark. Rn they technically swim a couple km to the enemy boat, climb on it, attack, and swim back :))

Reply:
You should probably check the target tile as being water instead of the unit type